steam/ext/dota2/client.py

```diff
@@ -0,0 +1,176 @@
+"""Licensed under The MIT License (MIT) - Copyright (c) 2020-present James H-B. See LICENSE"""
+
+from __future__ import annotations
+
+import asyncio
+from functools import partial
+from typing import Final
+
+from ..._const import timeout
+from ..._gc import Client as Client_
+from ...app import DOTA2
+from ...ext import commands
+from ...utils import MISSING
+from .enums import Hero
+from .models import LiveMatch
+from .protobufs.watch import (
+    CMsgClientToGCFindTopSourceTVGames,
+    CMsgGCToClientFindTopSourceTVGamesResponse,
+)
+from .state import GCState  # noqa: TCH001
+
+__all__ = (
+    "Client",
+    "Bot",
+)
+
+
+class Client(Client_):
+    """Represents a client connection that connects to Steam. This class is used to interact with the Steam API, CMs
+    and the Dota 2 Game Coordinator.
+
+    :class:`Client` is a subclass of :class:`steam.Client`, so whatever you can do with :class:`steam.Client` you can
+    do with :class:`Client`.
+    """
+
+    _APP: Final = DOTA2
+    _state: GCState  # type: ignore  # PEP 705
+
+    async def top_live_matches(
+        self,
+        hero: Hero = MISSING,
+        max_matches: int = 100,
+    ) -> list[LiveMatch]:
+        """Fetch top live matches
+
+        This is similar to game list in the Watch Tab of Dota 2 game app.
+        "Top matches" in this context means
+
+        * tournament matches
+        * highest average MMR matches
+
+        Parameters
+        ----------
+        hero_id
+            Filter matches by Hero.
+            Note, in this case Game Coordinator will still use only current top100 live matches, i.e. requesting
+            "filter by Muerta" will return only subset of those matches in which Muerta is currently being played.
+            It will not look into lower MMR games than top100. This behavior is consistent with how Watch Tab works.
+        max_matches
```

Comment:
```suggestion
        limit
```
Everywhere else calls this a limit so for consistency sake please keep it the same